### PR TITLE
Feat/lagged features names

### LINKED ISSUE #1670: I can't see the importances of my variables
Hi,

I just implemented a lightGBM. I would like to see the importances of my variables, but when I ask for feature importances I get an array of numbers, so i can't see which importance belongs to each variable.
image
And if i ask for the names of the variables so I can see the importances of each one, I get these names
image

So in the end, I'm still unable to view the importances of my variables.

In the other hand, I would like to know the criteria on which the importances are based. We don't expect to be equal the importances based on the criterion "split" than the ones based on the criterion "gain".
How can I solve these issues? Is there anything I can help with?

Thank you very much for your great job and for the attention.

### PR DESCRIPTION
**Pull request by madtoinou**
Fixes #1670.

### Summary

Added helper functions in the tabularization and regression model in order to generate labels for the lagged features (and static covariates if applicable), which are stored in the `RegressionModel.lagged_features_name_` attribute (`List[List[str]`). This enable the usage of the `feature_importances` attribute, available for some sklearn models. It was not possible before because the `create_lagged_data` method returns arrays and lose the name of the columns.

If the model was fit on a single `TimeSeries`, the attribute contains only one `List[str]`, if trained on a `Sequence` of `TimeSeries`, the attribute contains several `List[str]`.

The Lists containing the lagged features names are always nested, the API could eventually be simplified:
- if the components names are identical across all the `TimeSeries` used during training, retain the first one only
- wrapping the access to this attribute in a function, asking the user to provide a series (and mapping the training TimeSeries to their lagged features names)

### Additional information
Added the corresponding tests.

### CODE REVIEW

**Comment by madtoinou** [comment]
Updated the PR:
- if argument is a single `TimeSeries`, its components' name are used
- if argument is a `Sequence[TimeSeries]` and all the ts have exactly identical components names, these names are used
- if argument is a `Sequence[TimeSeries]` and any of the ts contains a component with a different name, generic names are generated ('target', 'past_cov', 'future_cov').

Each "variate" (target, past_covariates and future_covariates) is processed independently: it's possible to have a mixture of generic names and "original names".

**Comment by madtoinou** [comment]
Updated to use the same naming conventions as the explainability module: `{comp_name}_lag{lag_idx}` for unique component names, `comp{i}_{variate_type}_lag{lag_idx}` for generic names.

**Review by dennisbader — APPROVED** [review]
Looks good, thanks a lot @madtoinou 🚀 

Ready for release! 🥳
